File: client/src/app/shared/models/matricula.ts
import { Aluno } from "./aluno";
import { Curso } from "./curso";

export interface Matricula {
    id: number;
    numero: string;
    cursoId: number;
    alunoId: number;
    curso: Curso;
    aluno: Aluno; 
}

export interface InserirAlunoComMatriculaRequest {
    id: number;
    alunoId: number;
    cursoId: number;
  }

File: server/edusys.Api/Entities/Matricula.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace edusys.Api.Entities
{
    [Table("Matricula")]
    public class Matricula
    {
        [Key]
        public int Id { get; set; }
        public string? Matricula { get; set; }
        public int? AlunoId { get; set; }
        public virtual Aluno? Aluno { get; set; }
        public int? CursoId { get; set; }
        public virtual Curso? Curso { get; set; }        

    }
}
